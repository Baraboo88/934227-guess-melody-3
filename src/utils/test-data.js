export const welcomeScreenMockData = {
  numberOfErrors: 4,
  onStartButtonClick: () => {}
};

export const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [{
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `blues`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `jazz`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }],
  },
  {
    type: `artist`,
    song: {
      artist: `Jim Snow`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    },
    answers: [{
      picture: `https://api.adorable.io/avatars/128/0.6750575244607682`,
      artist: `John Daniels`,
    }, {
      picture: `https://api.adorable.io/avatars/128/0.4076094808626085`,
      artist: `Jack Beam`,
    }, {
      picture: `https://api.adorable.io/avatars/128/0.6972088175637985`,
      artist: `Jim Snow`,
    }],
  }
];

export const userAnswer = [false, false, false, false];
