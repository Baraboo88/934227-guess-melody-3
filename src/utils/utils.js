import {actionCreator} from "../reducer";

export const mapStateToProps = (state, props) => {
  if (state.step >= state.questions.length) {
    props.history.push(`/`);
  }
  const id = props.match.params.id;

  let question = state.questions.find((newOffer) => newOffer.id === Number(id));

  if (!question) {
    props.history.push(`/`);
  }
  const nextQuestion = state.questions[state.step + 1] ? {nextType: state.questions[state.step + 1].type, nextId: state.questions[state.step + 1].id} : {nextType: null, nextId: null};
  return {
    question,
    mistakes: state.mistakesNumber,
    nextType: nextQuestion.nextType,
    nextId: nextQuestion.nextId
  };
};

export const mapDispatchToProps = (dispatch) => ({
  onAnswer(question, answer) {
    dispatch(actionCreator.incrementMistake(question, answer));
    dispatch(actionCreator.incrementStep());
  }
});
