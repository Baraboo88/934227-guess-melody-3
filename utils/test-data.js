export const mockData = {
  numberOfErrors: 4
};
