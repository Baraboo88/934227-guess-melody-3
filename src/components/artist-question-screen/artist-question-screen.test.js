import renderer from 'react-test-renderer';
import React from 'react';
import ArtistQuestionScreen from "./artist-question-screen";
import {questions} from "../../utils/test-data";


it(`ArtistQuestionScreen successfully rendered`, () => {
  const tree = renderer.create(<ArtistQuestionScreen question={questions[1]} onAnswer={() => {}} activePlayerId = {0} onPlayButtonClick = {() => {}}/>);
  expect(tree).toMatchSnapshot();
});
