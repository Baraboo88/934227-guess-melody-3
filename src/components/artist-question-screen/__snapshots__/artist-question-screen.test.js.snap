// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen successfully rendered 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      />
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          data-test="test-answer-button"
          id="artist-0"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="John Daniels"
        />
        <label
          className="artist__name"
          htmlFor="artist-0"
        >
          <img
            alt="Пелагея"
            className="artist__picture"
            src="https://api.adorable.io/avatars/128/0.6750575244607682"
          />
          John Daniels
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          data-test="test-answer-button"
          id="artist-1"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="Jack Beam"
        />
        <label
          className="artist__name"
          htmlFor="artist-1"
        >
          <img
            alt="Пелагея"
            className="artist__picture"
            src="https://api.adorable.io/avatars/128/0.4076094808626085"
          />
          Jack Beam
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          data-test="test-answer-button"
          id="artist-2"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="Jim Snow"
        />
        <label
          className="artist__name"
          htmlFor="artist-2"
        >
          <img
            alt="Пелагея"
            className="artist__picture"
            src="https://api.adorable.io/avatars/128/0.6972088175637985"
          />
          Jim Snow
        </label>
      </div>
    </form>
  </section>
</section>
`;
